From f763216136cc102716731dc16246962b06735b45 Mon Sep 17 00:00:00 2001
From: "Jose M. Guisado Gomez" <guigom@riseup.net>
Date: Sun, 26 Jul 2020 11:56:59 +0200
Subject: [PATCH] nftables plugin: improve variable names about comments

There is a main distiction between comments, the one that is read from
collectd config section of this plugin, used for filtering purposes. Eg:

<Plugin nftables>
	inet filter input "in"
			  ^^^^
</Plugin>

User data comment (udata_comment) refers to the comment that is extracted from a rule
fetched using libnftnl (comments are stored inside user data of the
rule). The existance of a rule depend on the ruleset of the computer
running this.

Also, don't check for udata_comment to be NULL because free(ptr) does nothing
if ptr is NULL. nftnl_rule_get_comment will return NULL if the rule
being processed from the ruleset has no comment inside its user data.
---
 src/nftables.c | 17 ++++++++---------
 1 file changed, 8 insertions(+), 9 deletions(-)

diff --git a/src/nftables.c b/src/nftables.c
index 7ba38b4c1a..db50cde952 100644
--- a/src/nftables.c
+++ b/src/nftables.c
@@ -158,8 +158,8 @@ static int submit_cb(struct nftnl_expr *e, void *data) {
 
 static int table_cb(const struct nlmsghdr *nlh, void *data) {
   struct nftnl_rule *t;
-  char *comment = (char *)data;
-  char *t_comment = NULL;
+  char *filter_comment = (char *)data; /* config filter comment */
+  char *udata_comment = NULL;   /* rule userdata comment */
 
   t = nftnl_rule_alloc();
   if (t == NULL) {
@@ -172,20 +172,19 @@ static int table_cb(const struct nlmsghdr *nlh, void *data) {
     goto err_free;
   }
 
-  t_comment = nftnl_rule_get_comment(t);
+  udata_comment = nftnl_rule_get_comment(t);
   plugin_log(LOG_DEBUG, "table_cb | filter_comment: %s rule_comment: %s",
-             comment, t_comment);
-  if (strlen(comment) > 0) {
-    if (t_comment && strcmp(t_comment, comment) == 0) {
+             filter_comment, udata_comment);
+  if (strlen(filter_comment) > 0) {
+    if (udata_comment && strcmp(udata_comment, filter_comment) == 0) {
       nftnl_expr_foreach(t, submit_cb, t);
     }
-  } else if (t_comment) {
+  } else if (udata_comment) {
     nftnl_expr_foreach(t, submit_cb, t);
   }
 
 err_free:
-  if (t_comment)
-    free(t_comment);
+  free(udata_comment);
   nftnl_rule_free(t);
 err:
   return MNL_CB_OK;
